modules/grandpa/src/lib.rs

```diff
@@ -82,6 +90,24 @@ pub mod pallet {
 		/// until the request count has decreased.
 		#[pallet::constant]
 		type MaxRequests: Get<u32>;
+
+		/// The maximum length of a session on the bridged chain.
+		///
+		/// The pallet uses this to bound justification verification since justifications contain
+		/// ancestry proofs whose size is capped at `BridgedSessionLength`.
+		#[pallet::constant]
+		type BridgedSessionLength: Get<BridgedBlockNumber<Self, I>>;
+
+		/// The number of validators on the bridged chain.
+		///
+		/// The pallet uses this to bound justification verification since justifications may
+		/// contain up to `BridgedValidatorCount` number of signed `pre-commit` messages which need to
+		/// be verified.
+		#[pallet::constant]
+		type BridgedValidatorCount: Get<u64>;
```

Comment:
Why isn't it `u32` to prevent casting?

Reply:
> This is quite misleading right now, cause the validator count can be easily changed on the bridged chain and we should really be able to support not significant changes without updating the bridge setup.
>
> Like in case we set this to close to actual values we risk stalling the bridge, so it needs at least 2x buffer.

Good point, I hadn't thought about that